src/System.IO.Pipelines/PipelineWriterExtensions.cs

```diff
@@ -19,12 +17,14 @@ public static Task WriteAsync(this IPipeWriter output, Span<byte> source)
         private static Task FlushAsync(WritableBuffer writeBuffer)
         {
             var awaitable = writeBuffer.FlushAsync();
-            if (awaitable.IsCompleted)
+            if (awaitable.IsCompletedSuccessfully)
             {
-                awaitable.GetResult();
-                return _completedTask;
+                awaitable.GetResult(); // Is this needed?
```

Comment:
If `GetResult()` is throwing an error then you will need to catch it and return a faulted task, no?